pkg/ingester/user_tsdb.go

```diff
@@ -245,9 +246,11 @@ func (u *userTSDB) blocksToDelete(blocks []*tsdb.Block) map[ulid.ULID]struct{} {
 	shippedBlocks := u.getCachedShippedBlocks()
 
 	result := map[ulid.ULID]struct{}{}
-	for shippedID := range shippedBlocks {
-		if _, ok := deletable[shippedID]; ok {
-			result[shippedID] = struct{}{}
+	for blockID, blockShippedTime := range shippedBlocks {
+		if blockShippedTime.Before(time.Now().Add(-u.blockMinRetention)) {
```

Comment:
I'd extract `time.Now().Add(-u.blockMinRetention)` to a variable called `deadline` or similar.